### DISCUSSION TITLE
Delete Last Spawned vehicle.

### DISCUSSION THREAD

**Discussion by SlavicBlin** [tupoy-ya/TupoyeMenu#85] [Q&A] [answered]
Yo not sure if you can help since you said you dont really know how to add features with the whole new lua thing but i thought i would ask anyways.
I have been trying to add a check box like " Spawn Maxed " but instead one that says " Delete Last " now before this used to work by simply using the code provided in https://github.com/YimMenu/YimMenu/issues/1369#issuecomment-1560376506 however on the latest builds of both Yimmenu and Tupoyemenu i have not managed to get it working, i have tried a few methods as shown in this discussion https://github.com/YimMenu/YimMenu/discussions/2512 but still no progress all i could get it to do is delete any spawned vehilce which is not what i am trying to do.

**Comment by tupoy-ya** [comment]
###### I can't really understand what your problem is sorry.

> Yo not sure if you can help since you said you don't really know how to add features with the whole new lua thing but i thought i would ask anyways.

Do you want it to be a Lua script? Because in the links provided you use c++ code, I'm going to assume you use c++ in this reply.
Nothing below was tested.

Why are you using the `VEHICLE::DELETE_VEHICLE` native when there is a wrapper for it in YimMenu `entity::delete_entity` which _should_ work with vehicles.

Also there is the `PLAYER::GET_PLAYERS_LAST_VEHICLE` native which may work.

I think the `VEHICLE::GET_LAST_DRIVEN_VEHICLE` native you use has to be explicitly set by `VEHICLE::SET_LAST_DRIVEN_VEHICLE`, so you can:
1. Try using the `PLAYER::GET_PLAYERS_LAST_VEHICLE` native.
2. Use `VEHICLE::SET_LAST_DRIVEN_VEHICLE` when spawning new vehicles.
3. Store last vehicle id when you spawn it and then delete it when you want.

**Reply by SlavicBlin** [reply]
alright yeah i will try that.

**Reply by SlavicBlin** [reply, marked as answer]
You are actually a goat <3 i replaced `VEHICLE::GET_LAST_DRIVEN_VEHICL` with `PLAYER::GET_PLAYERS_LAST_VEHICLE` and `VEHICLE::DELETE_VEHICLE` with `entity::delete_entity` and it is working like a charm now <3 thank you so much
